cpp/oneapi/dal/backend/primitives/selection/kselect_by_rows_single_col.hpp

```diff
@@ -0,0 +1,158 @@
+                           const ndview<Float, 2>& data,
+                           std::int64_t k,
+                           ndview<std::int32_t, 2>& indices,
+                           const event_vector& deps) override {
+        ndarray<Float, 2> dummy;
+        return select<false, true>(queue, data, dummy, indices, deps);
+    }
+
+private:
+    template <bool selection_out, bool indices_out>
+    sycl::event select(sycl::queue& queue,
+                       const ndview<Float, 2>& data,
+                       ndview<Float, 2>& selection,
+                       ndview<std::int32_t, 2>& indices,
+                       const event_vector& deps = {}) {
+        if (indices_out) {
+            ONEDAL_ASSERT(indices.get_shape()[0] == data.get_shape()[0]);
+            ONEDAL_ASSERT(indices.get_shape()[1] == 1);
+        }
+        if (selection_out) {
+            ONEDAL_ASSERT(selection.get_shape()[0] == data.get_shape()[0]);
+            ONEDAL_ASSERT(selection.get_shape()[1] == 1);
+        }
+        const auto sg_sizes = queue.get_device().get_info<sycl::info::device::sub_group_sizes>();
+        ONEDAL_ASSERT(!sg_sizes.empty());
+
+        auto result = std::max_element(sg_sizes.begin(), sg_sizes.end());
+        ONEDAL_ASSERT(result != sg_sizes.end());
+
+        const std::int64_t sg_max_size = static_cast<std::int64_t>(*result);
+
+        const std::int64_t col_count = data.get_dimension(1);
+        const std::int64_t row_count = data.get_dimension(0);
+        const std::int64_t stride = data.get_shape()[1];
+
+        const std::int64_t row_adjusted_sg_num =
+            col_count / sg_max_size + (std::int64_t)((bool)(col_count % sg_max_size));
+        const std::int64_t expected_sg_num =
+            std::min(kselect_by_rows_single_col::preffered_wg_size / sg_max_size,
+                     row_adjusted_sg_num);
+        ONEDAL_ASSERT(expected_sg_num > 0);
+
+        const std::int64_t wg_size = expected_sg_num * sg_max_size;
+        sycl::range<2> global(wg_size, row_count);
+        sycl::range<2> local(wg_size, 1);
+        sycl::nd_range<2> nd_range2d(global, local);
+
+        const Float* data_ptr = data.get_data();
+        [[maybe_unused]] Float* selection_ptr =
+            selection_out ? selection.get_mutable_data() : nullptr;
+        [[maybe_unused]] std::int32_t* indices_ptr =
+            indices_out ? indices.get_mutable_data() : nullptr;
+        auto fp_max = detail::limits<Float>::max();
+
+        auto event = queue.submit([&](sycl::handler& cgh) {
+            cgh.depends_on(deps);
+            cgh.parallel_for(
+                make_multiple_nd_range_2d({ wg_size, row_count }, { wg_size, 1 }),
+                [=](sycl::nd_item<2> item) {
+                    auto sg = item.get_sub_group();
+                    const uint32_t sg_id = sg.get_group_id()[0];
+                    const uint32_t wg_id = item.get_global_id(1);
+                    const uint32_t sg_num = sg.get_group_range()[0];
```

Comment:
Can it be rewritten like this?
```suggestion
                    const uint32_t sg_num = sg.get_group_range(0);
```

Reply:
According to SYCL 2020 @https://www.khronos.org/registry/SYCL/specs/sycl-2020/pdf/sycl-2020.pdf,
such an overload doesn't exist for sycl::sub_group class.